### DISCUSSION TITLE
How to wrap TextNode/ElementNode HTMLElements with React?

### DISCUSSION THREAD

**Discussion by yf-yang** [facebook/lexical#5177] [Q&A]
I'm trying to rewrite the drag and drop functionality with dnd-kit, which is a React library and all the draggable should be React elements. I've checked Lexical's source code, it seems those TextNodes and ElementNodes's DOM is created and added to their parent DOM via vanilla JavaScript. What can I do if I want to control those nodes with a React library?

**Comment by moy2010** [comment]
In the [documentation](https://lexical.dev/docs/concepts/nodes) is mentioned that only `decorator nodes` can be used with frameworks such as React.

**Reply by yf-yang** [reply]
Well, we are not talking about the same question.

**Comment by frankdiw** [comment]
Did you solve it?
